test/distributed/rpc/faulty_agent/test_rpc_spawn.py

```diff
@@ -0,0 +1,19 @@
+#!/usr/bin/env python3
```

Comment:
Could you please explain how does `test_rpc_spawn.py` work with `test_faulty_agent_rpc_trap.py`? When I launch `py.test test/distributed/rpc/faulty_agent/test_rpc_spawn.py` when does the trap kick in?

Reply:
The trap actually doesn't kick in for OSS -  I think we need some sort of additional `setUp` function just for `FaultyAgentRpcTest` that sets the `TEST_CONFIG.build_rpc_backend_options` and `TEST_CONFIG.rpc_backend_name` for the faulty agent. What would be the best way of adding an additional setup step there?

The design for the trap is similar to Thrift internally, where the Buck rule for faulty agent tests defines the trap as the main module, which sets the right configs before running the tests in `FaultyAgentRpcTest`.

Reply:
> What would be the best way of adding an additional setup step there?

For long term, we might need a little overhaul of our test code structure to support additional backend types. For short term, we can have a `FaultyRpcAgentTestFixture` that inherit from `RpcAgentTestFixture` and override `rpc_backed` and `rpc_backend_options`? Then we let the `FaultAgentRpcTest` to inherit from `FaultyRpcAgentTestFixture`?

Above won't modify `TEST_CONFIG` default values, but should return the correct value for things like `self.rpc_backend` and `self.rpc_backend_options`, which should be OK as the plan for now (IIUC) is not inheriting all tests from RpcTest yet.

If we also need to modify `TEST_CONFIG`, we can do that in `faulty_agent/test_rpc_spawn.py`? More specifically, we can do that in `FaultyRpcAgentTestWithSpawn.setUp`?